src/display/annotation_layer.js

```diff
@@ -2627,23 +2757,44 @@ class AnnotationLayer {
 
   #annotationCanvasMap = null;
 
-  #div = null;
-
   #editableAnnotations = new Map();
 
-  constructor({ div, accessibilityManager, annotationCanvasMap }) {
-    this.#div = div;
+  constructor({
+    div,
+    accessibilityManager,
+    annotationCanvasMap,
+    l10n,
+    page,
+    viewport,
+  }) {
+    this.div = div;
     this.#accessibilityManager = accessibilityManager;
     this.#annotationCanvasMap = annotationCanvasMap;
+    this.l10n = l10n;
+    this.page = page;
+    this.viewport = viewport;
+    this.zIndex = 0;
+
+    if (typeof PDFJSDev === "undefined" || PDFJSDev.test("TESTING")) {
+      // For testing purposes.
+      Object.defineProperty(this, "togglePopups", {
+        value: () => {
+          for (const toggle of this.popupToggle) {
+            toggle();
+          }
+        },
+      });
+      this.popupToggle = [];
```

Comment:
Will this actually do the right thing in general, since if the Popup has a `/Open true` entry we're now opening it by default and I believe that this method would incorrectly *close* the Popup in such a case?

It seems to me that we rather want a method called e.g. `showPopups` that calls the private `#show`-methods instead to prevent issues here?

Reply:
That's correct.